### PR TITLE
refactor core structure

### PR DESCRIPTION
**Pull request by miltontony**

### CODE REVIEW

**Comment by miltontony** [comment]
@smn I'm going to land this - the next PR will add a Docker controller and should be a lot clearer on the approach I'm taking

File: .coveragerc [hunk 1]
```diff
@@ -1,2 +1,2 @@
 [run]
-omit = unicoremc/migrations/*,ve/*
+omit = controllers/base/migrations/*,organizations/migrations/*,ve/*
```
Comment (smn) [inline .coveragerc]: I think you can do `**/migrations/*` here and match all subdirs.

File: controllers/base/forms.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    marathon_instances = forms.IntegerField(
+        required=False,
+        min_value=0,
+        widget=forms.TextInput(attrs={'class': 'input-small'}))
+
+    class Meta:
+        model = Controller
+        fields = (
+            'name', 'marathon_cpus', 'marathon_mem', 'marathon_instances',
+            'marathon_cmd')
```
Comment (smn) [inline controllers/base/forms.py]: Where is `marathon_cmd` defined?

Comment (smn) [inline controllers/base/forms.py]: Ah it's already on the model.

File: controllers/base/namers.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import random
+
+
+def do_me_a_unique_slug(model_cls, field):
+    """
+    model_cls: A Django model to enforce uniqueness
+    field: The field that stores the unique name
+    """
```
Comment (smn) [inline controllers/base/namers.py]: `do_me_a_slug()`? lol :D

**Comment by smn** [comment]
:+1: just left minor comments mostly because I was reading to  see what had changed, pls ignore :)

**Pull request merged**